server/models/tags/getTag.js

```diff
@@ -0,0 +1,27 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {Object} param0.query
+ * @return {Promise}
+ */
+
+function getTag({ query }) {
+  return new Promise((resolve, reject) => {
+    const keyword = query.keyword || ''
+    const searchId = query.id || ''
+    pool.query(
+      `SELECT * FROM tags WHERE id LIKE ? AND name LIKE ?`,
```

Comment:
Actually I meant to say:
```
let query = ``;
const depArr = [];

if(searchId) {
 query += `SELECT * FROM tags WHERE id = ?`;
 depArr.push(searchId);
} else if(keyword) {
 query += `SELECT * FROM tags LIKE ?`;
 depArr.push(keyword);
}

pool.query(query, depArr, (err, res) => {
 ...
})

Reply:
Yeah, that's much better !

Reply:
complexity will be same in both cases....

Reply:
Nope ! this would be a bit faster , as it doesn't have AND , OR in SQL query..